FWCore/ParameterSet/python/Config.py

```diff
@@ -422,9 +422,9 @@ def __setattr__(self,name,value):
         if not name.replace('_','').isalnum():
             raise ValueError('The label '+name+' contains forbiden characters')
 
-        if name == 'options':
+        if name == 'options' and hasattr(self,name):
             value = self.__updateOptions(value)
-        if name == 'maxEvents':
+        if name == 'maxEvents' and hasattr(self,name):
```

Comment:
The purpose and effect of these changes is not clear to me (even after reading their commit message). Could you elaborate?

Reply:
The constructor of `Process` has the line
```python
        self.options = Process.defaultOptions_()
        self.maxEvents = Process.defaultMaxEvents_()
```
Both of those trigger the `__setattr__` call. As the attributes have not been called yet, it is more efficient not to call the `__update*` methods and instead just set the attribute directly.

Reply:
What happens (or should happen) in the following cases?
```py
process = cms.Process("Test")
# 1
del process.options
process.options = cms.PSet(numberOfThreads = cms.untracked.uint32(5))
# 2
del process.options
process.options = cms.PSet(nonExistent = 6)
# 3
del process.options
process.options = dict(numberOfThreads = 7)
# 4
del process.options
process.options = dict(nonExistent = 8)
```